qualang_tools/simulator/simulator.py

```diff
@@ -0,0 +1,52 @@
+import numpy as np
+from qm import ControllerConnection, InterOpxChannel
+
+
+def create_simulator_controller_connections(
+    n_controllers, actual_controllers=None, n_connections=12, print_debug=False
+):
+    """
+    Creates a list of :class:`~qm.simulate.interface.ControllerConnection` objects with the Inter-controllers
+    connections between controllers in a way which maximizes the optical links connectivity
+    For use inside the :class:`~qm.simulate.interface.SimulationConfig` for simulating the controllers` behavior.
+
+    :param n_controllers: The number of controllers
+    :type n_controllers: int
+    :param actual_controllers: A list of the controllers actually being used in the simulation
+    :type actual_controllers: List[int]
+    :param n_connections: The number of optical connectors each controller has, default is 12
+    :type n_connections: int
+    :param print_debug: If true, prints the connections to the terminal
+    :type print_debug: bool
+    :returns: The ControllerConnection object
+
+    """
+    if not actual_controllers:
+        actual_controllers = [i + 1 for i in range(n_controllers)]
+    controller_connections = []
+    unused_connection = np.ones((n_controllers, n_connections), dtype=bool)
+    while unused_connection.any():
+        for i in range(n_controllers):
+            for j in range(i + 1, n_controllers):

```

Comment:
if n_controllers is 1 and actuall_controllers is not None as in `create_simulator_controller_connections(1, [1]),` then the script will hang forever. I know it is not meant to happen, but maybe raising an error/warning could avoid some customer issues :)

Reply:
Made it to return an empty list, which is also the expected output